src/AppInstallerCommonCore/Public/winget/MSStoreDownload.h

```diff
@@ -12,76 +14,43 @@
 
 namespace AppInstaller::MSStore
 {
-    enum class PackageFormatEnum
+    // Struct representing 1 MSStore package file download info
+    struct MSStoreDownloadFile
     {
-        Unknown = -1,
-        AppxBundle,
-        EAppxBundle,
-        MsixBundle,
-        Appx,
-        Msix,
+        std::string Url;
+        std::vector<BYTE> Sha256;
+        std::string FileName;
+        AppInstaller::Utility::UInt64Version Version = 0;
     };
 
-    struct MSStoreCatalogPackage
+    struct MSStoreDownloadInfo
     {
-        std::string PackageId;
-
-        std::vector<AppInstaller::Utility::Architecture> Architectures;
-
-        std::vector<std::string> Languages;
-
-        PackageFormatEnum PackageFormat = PackageFormatEnum::Unknown;
-
-        std::string WuCategoryId;
-
-        std::string ContentId;
+        std::vector<MSStoreDownloadFile> MainPackages;
+        std::vector<MSStoreDownloadFile> DependencyPackages;
     };
 
-    namespace details
+    struct MSStoreDownloadContext
     {
-        struct MSStoreCatalogPackageComparisonField
-        {
-            MSStoreCatalogPackageComparisonField(std::string_view name) : m_name(name) {}
+        MSStoreDownloadContext(
+            std::string productId,
+            AppInstaller::Utility::Architecture architecture,
+            AppInstaller::Manifest::PlatformEnum platform,
+            std::string locale,
+            AppInstaller::Authentication::AuthenticationArguments authArgs);
 
-            virtual ~MSStoreCatalogPackageComparisonField() = default;
+        // Calls display catalog API and sfs-client to get download info.
+        MSStoreDownloadInfo GetDownloadInfo();
 
-            std::string_view Name() const { return m_name; }
-
-            virtual bool IsApplicable(const MSStoreCatalogPackage& package) = 0;
-
-            virtual bool IsFirstBetter(const MSStoreCatalogPackage& first, const MSStore::MSStoreCatalogPackage& second) = 0;
-
-        private:
-            std::string_view m_name;
-        };
-    }
-
-    // Class for comparing MSStore packages.
-    struct DisplayCatalogPackageComparator
-    {
-        DisplayCatalogPackageComparator(const std::vector<std::string>& requiredLanguages, const std::vector<AppInstaller::Utility::Architecture>& requiredArchs);
-
-        // Gets the best installer from the manifest, if at least one is applicable.
-        std::optional<MSStoreCatalogPackage> GetPreferredPackage(const std::vector<MSStoreCatalogPackage>& package);
-
-        // Determines if an installer is applicable.
-        bool IsApplicable(const MSStoreCatalogPackage& package);
-
-        //// Determines if the first installer is a better choice.
-        bool IsFirstBetter(const MSStoreCatalogPackage& first, const MSStoreCatalogPackage& second);
+        // Gets license for the corresponding package returned by previous GetDownloadInfo().
+        // GetDownloadInfo() must be called before calling this method.

```

Comment:
This comment suggests that this method should be on the download info object instead...